enhancements/dns/coredns-egressfirewall-integration.md

```diff
@@ -0,0 +1,363 @@
+---
+title: coredns-egressfirewall-integration
+authors:
+  - '@arkadeepsen'
+reviewers:
+  - '@Miciah'
+  - '@danwinship'
+  - '@JoelSpeed'
+  - '@TrilokGeer'
+approvers:
+  - '@Miciah'
+  - '@danwinship'
+api-approvers:
+  - '@JoelSpeed'
+creation-date: 2023-01-31
+last-updated: 2023-01-31
+tracking-link:
+  - https://issues.redhat.com/browse/CFE-748
+see-also:
+  - N.A.
+replaces:
+  - N.A.
+superseded-by:
+  - N.A.
+---
+
+# Improve CoreDNS Integration with EgressFirewall
+
+## Summary
+
+This enhancement improves the integration of CoreDNS with EgressFirewall. With this improved
+integration, EgressFirewall will be able to better support DNS names whose IPs change dynamically
+and also will be able to provide support for wildcard DNS names.
+
+## Motivation
+
+Currently, EgressFirewall (OVN-K master) does a DNS lookup of a DNS name based on a default TTL or the
+TTL of the previous lookup. OVN-K master then updates the underlying ``AddressSet`` for the DNS name
+referenced by the corresponding ACL rule of the EgressFirewall rule containing the DNS name. However,
+if a pod, belonging to the same Namespace as that of the EgressFirewall, does a DNS lookup and is able
+to get the response before the OVN-K master then the pod will be incorrectly allowed/denied access to
+the DNS name. The integration between CoreDNS and EgressFirewall needs to be improved to avoid such a
```

Comment:
```suggestion
if a pod, belonging to the same Namespace as that of the EgressFirewall, does a DNS lookup and gets a different response than the OVN-K master has, then the pod will be incorrectly allowed/denied access to
the host. The integration between CoreDNS and EgressFirewall needs to be improved to avoid such a
```